Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/RiskLevelRepository.kt

```diff
@@ -7,7 +7,8 @@ import kotlinx.coroutines.flow.MutableStateFlow
 
 object RiskLevelRepository {
 
-    private val internalRisklevelScore = MutableStateFlow(getLastSuccessfullyCalculatedScore().raw)
+    private val internalRisklevelScore = MutableStateFlow(if (getLastSuccessfullyCalculatedScore() !=
+        RiskLevel.UNDETERMINED) getLastSuccessfullyCalculatedScore().raw else RiskLevel.UNKNOWN_RISK_INITIAL.raw)
```

Comment:
I'd recommend moving this into a function call.

Also, getLastSuccessfullyCalculatedScore() is called twice, please "cache" it in a val.

Most importantly, I don't see how why this value has to be overwritten in this particular Flow initializer. getLastSuccessfullyCalculatedScore() solely accesses the LocalData. UNDETERMINED is a default value if no data have been found/stored or the risk value integer value cannot be mapped to a constant. IMHO this is not addressing the issue in the right place. How does this relate to the no-wifi problem?

Lastly, please ensure it does not break the fixed behavior of EXPOSUREAPP-3723.

Reply:
there were just some minutes left on Friday for trying to fix this. I'm going to optimize the code. Will also check Exposureapp-3723 for collisions. If it does, further investigations by the original author are required.

Reply:
I think it should be if(getLastSuccessfullyCalculatedScore().raw != RiskLevel.UNDETERMINED) getLastSuccesfullCalculatesScore.raw() else RiskLevelConstants.UNKNOWN_RISK_INITIAL

This more closely aligns with the original and the changes made here:
https://github.com/corona-warn-app/cwa-app-android/pull/1597/files

I can test when I get a chance

Reply:
Although I think they effectively do the same thing.

Reply:
same effect.